### DISCUSSION TITLE
clearErrors is not setting isValid to true

### DISCUSSION THREAD

**Discussion by anjana543** [react-hook-form/react-hook-form#4384] [Ideas]
Hi @bluebill1049,

We've developed a dynamic form that contains multiple select boxes, text, and textarea fields.

![image](https://user-images.githubusercontent.com/4394526/110330519-d8006200-8043-11eb-86e8-840cb9b0ad4f.png)

When the user changes the value of the end year, an error message should be displayed If the start date is lesser than the end date. For this I'm using controller rules validation function as below,
 <Controller
        as={<ReactSelect />}
        defaultValue={selectedValue}
        options={optionSet}
        className={className}
        name={name}
        isMulti={isMulti}
        isClearable={isClearable}
        isSearchable={isSearchable}
        control={control}
        placeholder={label}
        isLoading={loading}
        rules={{
          required: requiredData?.fields?.value?.value || false,
          **validate: () => dataValidator(),**
        }}

and in dataValidator function, based on the start and end date, I'm using **clearErrors** function to remove all validation from start month, start year and end month.

**Issue:** After using clearErrors(['startMonth', 'startYear', 'endMonth', 'endYear']), the formState.isValid is not set to true.

Please help me to identify the issue.

Using React: 16.15.1

Thanks.

**Comment by bluebill1049** [comment]
provide a csb

**Reply by anjana543** [reply]
@bluebill1049 Please check the below csb to reproduce the error,
https://codesandbox.io/s/hungry-monad-iow4z?file=/src/App.js

Steps to reproduce:
1. Change **Start month** as April and **End month**  as Jan
2. Check the isValid key after changing End month to Sept
3. isValid is not set to true after using clearErrors

Thank you.

**Reply by bluebill1049** [reply]
@andidev `clearErrors`  doesn't reset `isValid`, it will only clear the errors object. Your validation will pass before your form becomes valid, if you want to skip validation with certain conditions consider write a validate function with a conditional check for input validation.

**Comment by bluebill1049** [comment]
<img width="903" alt="Screen Shot 2021-03-10 at 10 26 47 pm" src="https://user-images.githubusercontent.com/10513364/110622555-bc39bf00-81ef-11eb-8113-cbf09d63f194.png">

**Reply by anjana543** [reply]
@bluebill1049 How can I validate my other fields? Any way to set isValid to true? 
Any alternative solution?
I tried using the watch function but unable to use SetErrors() inside the useeffect of watch function because it resulted in an infinite loop. Please help me here...

**Reply by bluebill1049** [reply]
@anjana543 please don't ping me multiple times... I understand your frustration and want to get this to work. I am here to help, if i see your question i will try to reply.

potential: include logic in your validate function to skip that validation.

```
validate: () => {
  if(xxx) return true; // you can use external state to resolve this
  // do the rest of your validation 
}
```

**Reply by anjana543** [reply]
Ok @bluebill1049
